### DISCUSSION TITLE
Configuring heap space

### DISCUSSION THREAD

**Discussion by eriklumme** [gunnarmorling/1brc#268] [General]
Most solutions do not seem to explicitly configure this.  On my machine, the default max heap space is 25% of the available RAM.

I seem to get the best result with this heap size, where my solution runs much slower if the heap is larger. With that in mind, it would make sense to configure.

One issue is that if your program runs better the more RAM it has, you can't configure a static value, if the RAM on the evaluation machine differs from your local machine. On the other hand, setting it as a percentage might not be ideal either, as the ratio of free memory to total memory could also differ.

Thoughts on this?

**Comment by eriklumme** [comment]
I found it interesting that my solution runs almost as fast with 512 BM of heap space as it does with 8 GB. Seems to be very much CPU-limited.

**Comment by sharpobject** [comment]
I suspect that most fast solutions will be fine with pretty small heaps because the cost of regularly touching a lot of auxiliary storage is high.

**Comment by yavuztas** [comment]
In my experience, my solution was always slower when I tried with more heap space. It's not dramatic though, only 50 - 100 ms. But it counts in this challenge for sure. And for smaller sizes, I found serial GC performs best. I didn't see any difference even with Epsilon GC. Maybe Serial GC is also not kicked in since the heap usage is really small.
